controller/state/statedb.go

```diff
@@ -591,6 +615,29 @@ func (s *stateDB) Update(ctx context.Context, taskID string, req tasks.UpdateTas
 	return task, nil
 }
 
+func (s *stateDB) log(ctx context.Context, task tasks.Task, tx *sql.Tx) {
+	finalized, err := task.Finalize(ctx, txContextStorer(ctx, tx))
```

Comment:
this is going to error if the task is not Successful / errored -- see https://github.com/filecoin-project/dealbot/blob/main/tasks/model.go#L348

is this what you want? It looks like you're trying to log all updates, but I think this is going to cause your code to skip updates that aren't to a success state.